File: objc/ChainOfResponsibility/ChainOfResponsibility/Divison.m
#import "Divison.h"

@implementation Divison {
    NSMutableArray<Divison *>* nextList;
}

-(instancetype)init {
    if(self = [super init]) {
        nextList = [[NSMutableArray alloc] init];
    }
    return self;
}

-(void)addNextDivison:(Divison*)divison {
    [nextList addObject:divison];
}

-(void)handle:(Request*)request {
    for (Divison* d in nextList) {
        [d handle:request];
    }
}

-(void) processRequest {
    // Do nothing
}

@end

File: swift/ChainOfResponsibility/ChainOfResponsibility/Divison.swift
import Foundation

public class Divison {
    
    var nextList = [Divison]()
    
    init() {
        
    }
    
    func addNextDivison(_ divison: Divison) -> Void {
        nextList.append(divison)
    }
    
    func handle(_ request: Request) -> Void {
        for d in nextList {
            d.handle(request)
        }
    }
    
    func processRequest() -> Void {}

}
